### PR TITLE
Support tagging nodes [sc-81509]

### PR DESCRIPTION
**Pull request by AFrancani**
Adds a `tags` parameter in the `node-pool-request` paramset to define a map of tags which are added to the EC2 instances that are created for the node group.

In order to test, add a node pool to an existing cluster and set values in the new `tags` input. Once the node group is created, navigate to your EC2 instances and verify the tags are correctly added to the instances of the node group.

[sc-81509]

### CODE REVIEW

**Review by marchelleboid — CHANGES_REQUESTED** [review]
There's the problem of the initial node group, you can't tag those instances even though the field appears in the UI. I'm guessing you maybe tested it out, but even adding the `--tags` arg to the `create cluster` command tags everything except the instances unfortunately.

I guess the two options if we want to support tagging the initial node group is to either manually tag the instances after the cluster creation, which seems like it might be pretty annoying, or to not create the initial nodegroup in the `create cluster` call, but to make a separate `create nodegroup` immediately after (then the code would probably be identical to the add node group runnable). That would make this PR quite a bit bigger though 😄

File: parameter-sets/node-pool-request/parameter-set.json [hunk 1]
```diff
@@ -74,6 +74,13 @@
             "description": "Enable GPU workloads on the cluster",
             "type": "BOOLEAN",
             "defaultValue": false
+        },
+        {
+            "name": "tags",
+            "label": "Tags",
+            "description": "Tags to apply on the created EC2 instances",
+            "type": "STRING",
```
[the remark this answers is not in the document]
Comment (AFrancani) [reply parameter-sets/node-pool-request/parameter-set.json:82]: Damn! That's what happen when you develop your plugin in DSS... 😅

**Comment by marchelleboid** [comment]
> There's the problem of the initial node group, you can't tag those instances even though the field appears in the UI. I'm guessing you maybe tested it out, but even adding the `--tags` arg to the `create cluster` command tags everything except the instances unfortunately.

Disregard, I missed that there were 2 pages of tags when using the AWS console 😅

**Review by marchelleboid — APPROVED** [review]
LGTM, initial nodes and added nodes have tags. I didn't test to see if nodes added because of the auto scaling policy are also tagged, but I assume that it's fine.

**Pull request merged**